### PR TITLE
Updates #545 - fixes User typeDef

### PR DESCRIPTION
**Pull request by kondanna**
type User {
    id: Int!
    username: String!
    userLesson: UserLesson
    email: String!
    name: String!
    isAdmin: Boolean! 
    cliToken: String
  }

### CODE REVIEW

File: graphql/typeDefs.ts [hunk 1]
```diff
@@ -112,11 +112,11 @@ export default gql`
...
     id: Int!
-    username: String
+    username: String!
     userLesson: UserLesson
-    email: String
-    name: String
-    isAdmin: String
+    email: String!
+    name: String!
+    isAdmin: Boolean!
```
Comment (ggwadera) [inline graphql/typeDefs.ts:119]: Until we make the `username`, `email` and `name` fields non-nullable on the database, this will conflict with the return types from prisma queries, and putting `as any` after it is not a good solution.

File: components/AppNav.tsx [hunk 2]
```diff
@@ -115,7 +115,7 @@ const AppNav: React.FC<{}> = () => {
   const { data } = useGetAppQuery()
   useEffect(() => {
     if (data && data.session) {
-      setSession(data.session)
+      setSession(data.session as Session)
```
Comment (Ulisseus) [inline components/AppNav.tsx:118]: Why is this needed? Apollo hook returns properly typed `GetAppQuery | undefined`, after check for `data&&data.session` typescript should infer proper type by itself.

Comment (kondanna) [reply components/AppNav.tsx:118]: Ah, ignore this PR. The more recent submissions and challenge one takes priority. 

I probably needed to add annotation because typescript was yelling at me and that was the only way to make it disappear

**Comment by kondanna** [comment]
see challengeType and submissionType PR. also starType will be updated first
